src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -1243,7 +1251,7 @@ private void transformMetadata() {
       }
       try {
         final String du = params.get(KEY_DISPLAY_URL);
-        if (!Strings.isNullOrEmpty(du) && !du.equals(origDisplayUrlStr)) {
+        if (!Strings.isNullOrEmpty(du)) {
```

Comment:
It's not your code, but it just prevents us from reparsing the URL if it was changed by the transform pipeline. Seems harmless enough and not worth changing.